src/Bicep.Core.UnitTests/Diagnostics/LinterRuleTests/SimplifyJsonNullRuleTests.cs

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Bicep.Core.Analyzers.Interfaces;
+using Bicep.Core.Analyzers.Linter.Rules;
+using Bicep.Core.UnitTests.Assertions;
+using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System.Linq;
+
+namespace Bicep.Core.UnitTests.Diagnostics.LinterRuleTests
+{
+    [TestClass]
+    public class SimplifyJsonNullRuleTests : LinterRuleTestsBase
+    {
+        private void ExpectPass(string text, Options? options = null)
+        {
+            AssertLinterRuleDiagnostics(SimplifyJsonNullRule.Code, text, diags => 
+            {
+                diags.Should().HaveCount(0, $"expecting linter rule to pass");
+            }, 
+            options);
+        }
+
+        private void ExpectDiagnosticWithFix(string text, string expectedFix)
+        {
+            AssertLinterRuleDiagnostics(SimplifyJsonNullRule.Code, text, diags => 
+            {
+                diags.Should().HaveCount(1, $"expected one fix per testcase");
+
+                diags.First().As<IBicepAnalyerFixableDiagnostic>().Fixes.Should().HaveCount(1);
+                diags.First().As<IBicepAnalyerFixableDiagnostic>().Fixes.First().Replacements.Should().HaveCount(1);
+                diags.First().As<IBicepAnalyerFixableDiagnostic>().Fixes.First().Replacements.First().Text.Should().Be(expectedFix);
+            });
+        }
+
+        [DataRow(
+            @"
+                param myParam object = {
+                    prop1: json('null')
+                }
+            ",
+            "null"
+        )]
+        [DataRow(
+            @"
+                var myVar = json('null')
+            ",
+            "null"
+        )]
+        [DataRow(
+            @"
+                resource stgAcct 'Microsoft.Storage/storageAccounts@2022-09-01' = {
+                    name: 'myStorageAccount'
+                    properties: {
+                        azureFilesIdentityBasedAuthentication: json('null')
+                    }
+                }
+            ",
+            "null"
+        )]
+        [DataRow(
+            @"
+                output myOutput object = {
+                    prop1: json('null')
+                }
+            ",
+            "null"
+        )]
+        [DataTestMethod]
+        public void Rule_ProducesDiagnosticWithFix(string text, string expectedFix)
+        {
+            ExpectDiagnosticWithFix(text, expectedFix);
+        }
+
+        [TestMethod]
```

Comment:
Could you add some more testcases here?  Including stuff like:

json(null)
json('nullx')
'json(\'null\')'
'''json('null')'''

Reply:
nit: Technically stuff like the following is valid and should probably be replaced:

var a = json(' null ')

Reply:
Added more test cases

Reply:
> nit: Technically stuff like the following is valid and should probably be replaced:
> 
> var a = json(' null ')

added `Trim()` to code fix check